sdk/storage/src/blob/blob/requests/get_blob_builder.rs

```diff
@@ -1,389 +1,170 @@
-                    request = LeaseIdOption::add_optional_header(&self, request);
-                    request = RangeOption::add_optional_header(&self, request);
-
-                    if r.len() <= 4 * 1024 * 1024 {
-                        request = request.header_static(RANGE_GET_CONTENT_MD5, "true");
-                    }
-                }
+                request = add_optional_header(&self.range, request);
+                request = add_optional_header(&self.client_request_id, request);
+                request = add_optional_header_ref(&self.lease_id, request);
                 request
             },
             None,
         )?;
 
-        let expected_status_code = if self.range().is_some() {
-            StatusCode::PARTIAL_CONTENT
+        let expected_status_code = if self.range.is_some() {
+            http::StatusCode::PARTIAL_CONTENT
         } else {
-            StatusCode::OK
+            http::StatusCode::OK
         };
 
-        let (headers, body) = perform_request_response
-            .check_status_extract_headers_and_body(expected_status_code)
+        let response = self
+            .blob_client
+            .http_client()
+            .execute_request_check_status(request, expected_status_code)
             .await?;
-        let blob = Blob::from_headers(&blob_name, &container_name, snapshot_time, &headers)?;
-        GetBlobResponse::from_response(&headers, blob, &body)
+
+        debug!("response.headers() == {:#?}", response.headers());
+
+        let blob = Blob::from_headers(
+            self.blob_client.blob_name(),
+            self.blob_client.container_client().container_name(),
+            response.headers(),
+        )?;
+        Ok(GetBlobResponse::from_response(
+            response.headers(),
+            blob,
+            response.body(),
+        )?)
+    }
+
+    pub fn stream(
+        self,
+        chunk_size: u64,
+    ) -> impl Stream<Item = Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>>> + 'a {
+        enum States {
+            Init,
+            Progress(Range),
+            End,
+        }
+
+        let complete_range = Range::new(0, u64::MAX);
+
+        futures::stream::unfold(States::Init, move |state| async move {
+            let remaining = match state {
+                States::Init => self.range.unwrap_or(complete_range),
+                States::Progress(range) => range,
+                States::End => return None,
+            };
+
+            let range = if remaining.start + chunk_size > remaining.end {
+                Range::new(remaining.start, remaining.end)
+            } else {
+                Range::new(remaining.start, remaining.start + chunk_size)
+            };
+
+            let req = self.range(range);
+
+            let response = match req.execute().await {
+                Ok(response) => response,
+                Err(err) => return Some((Err(err), States::End)),
+            };
+
+            Some((
```

Comment:
nit: unpacking that this huge value so that it's not a ~30 line expression might make things easier to read.